### ISSUE TITLE
D7 beta accessibility: homepage

### ISSUE THREAD

**Issue by jechols** [DSpace/dspace-angular#610]
Sorry in advance for the wall of text.  I've run an Accessibility Insights "guided" report against [the DSpace 7 beta homepage](https://dspace7-demo.atmire.com/home).  This was mostly for our needs (University of Oregon) to see what kinds of gaps we'd be looking at if D7 were released without further improvements, but it may be helpful to others working on the UI.

Unfortunately I don't have time to do much more than this (it took nearly a year for me to even find time to get this much done) until UO actually gets ready to adopt DSpace 7.

I provided the raw Accessibility Insights report, but I've tried to break things down in this ticket because those reports are a bit painful to parse sometimes.  Note that the automated scan's failures aren't listed below, as those are easy for anybody to run themselves even if the attached report isn't helpful.

[Assessment_20200305_DSpaceAngularHome.zip](https://github.com/DSpace/dspace-angular/files/4295365/Assessment_20200305_DSpaceAngularHome.zip)
---

## NVDA Testing

I only did a very brief test with NVDA and Firefox, but clicking/activating
links doesn't seem to work - the new page isn't announced in any way more often
than it is.  I suspect Angular is preventing NVDA from knowing the page changed
since the change is only to the DOM, and not an actual page load.

## Keyboard

Navigation:

- "All of DSpace" dropdown doesn't seem able to be activated by keyboard
- Can't display search field with keyboard
- Language selection can be activated by keyboard, but its dropdown isn't reachable

## Focus

Visible Focus:

- "Search" icon doesn't display focus

Modal dialogs:

- User can tab away from the login dialog

## Landmarks

Navigation region shouldn't have a search box - that should be a search region.
The nav region here could probably just be removed anyway, since the only
non-search things left are actions more than a way to get around.

## Repetitive content

Skipped some of this as I'm only testing the homepage for now

## Links

Link function: all failed links should just be buttons

Link purpose: Various links have missing a11y data:

- search "link" has no a11y info
- DSpace logo has no a11y info
- Statistics link goes to "/home" - probably just an oversight

## Native widgets

Instructions and Expected input: Search field in upper-right has no a11y info

## Custom widgets

**Note**: *Accessiblity Insights didn't examine the search button, login button, or
"All of DSpace" button.  These are all acting as custom widgets and need to be
coded as such, but AI can't detect them as custom widgets because they look
like plain links to AI.*

Widget information is defined in the [W3C reference for ARIA
practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

Design pattern: the language button should be coded as a menu

Role, state, property: the language button doesn't define an `aria-controls`
attribute - this isn't a failure, but would be beneficial

Keyboard Interaction: the language button and menu aren't implementing any of
the keyboard controls necessary for a menu

## Errors / status

Skipped - not familiar enough to comment on this section

## Parsing

Two occurrences of "weird" elements under `<ul>` tags

## Images

Image function: DSpace logos on top and bottom of page both need to be coded as
meaningful since they're links

## Language

No page-level language defined, and no language around things like "Aqui dspace
Brasil", though that one may be due to the content language not being properly
defined by the submitter, I really don't know.

## Text Legibility

Contrast fails for Statistics link - AI wasn't able to determine this
programmatically

Reflow fails: the "All of DSpace" submenu items get chopped off, and the page
doesn't seem to work anymore in various odd ways

## Semantics

CSS content: I'm not sure about the icons - seems like the search button has
issues here, but it's not 100% clear to me.

## Contrast

Search icon's lack of focus means it technically passes contrast: there's no
visual indicator to look at contrast.  This just means when it does have a
proper focus indicator, it needs to have its contrast double-checked.

**Comment by tdonohue** [comment]
Thanks for sharing this report, @jechols !  I've shared this in Slack to see if we can find a few folks willing to chip in on fixing these little by little.  Most seem smallish (in that they should be relatively easy to fix), but others need a bit more analysis as you note.

**Comment by tdonohue** [comment]
Closing as this is very outdated, and (at a glance) many/most of these appear to now be solved.
